Entity/Dtos/ProductDetailDto.cs

```diff
@@ -1,16 +1,31 @@
-﻿using WebNetSample.Core.Entities;
+﻿using Microsoft.AspNetCore.Http;
+using WebNetSample.Core.Entities;
 
 namespace WebNetSample.Entity.Dtos;
 
-public class ProductDetailDto : BaseDto
+public record ProductDetailDto : BaseDto
 {
-    public string Name { get; set; }
+    public ProductDetailDto(
+    string name,
```

Comment:
I advise you to add tabulation here to make difference between args and constructor declaration move obvious